testing/integration/monitoring_logs_test.go

```diff
@@ -111,21 +98,22 @@ func (runner *EnrollRunner) TestEnroll() {
 	require.NotZero(t, len(docs.Hits.Hits))
 	t.Logf("metricbeat: Got %d documents", len(docs.Hits.Hits))
 
-	// Stage 4: make sure we have no errors
-	t.Log("Making sure there are no error logs")
-	docs = findESDocs(t, func() (tools.Documents, error) {
-		return tools.CheckForErrorsInLogs(info.ESClient, info.Namespace, []string{})
```

Comment:
What were the error logs this was failing on? Should they exist or are we just ignoring a bug with this? It may be better to filter out known false positives than to remove this. 

There have been a lot of problems in the logs that are hidden because we have no automation that can detect them, this is one of the few places that can do that for us.

I am worried we are "fixing" the test by making it a worse test.

Reply:
It's not the case here. The error logs were from the agent trying to connect to the default ES, `localhost:9200`. Besides using the health of the agent and its components makes the test better, the test won't be affected by changes in the log messages and transient/recoverable errors.

This tests want to ensure the agent enrolls and the monitoring works. If any issue happens and the agent is capable of recovering and reaches the final desired state, it's not an issue. It shows the "self-healing" strategies we have work. Also taking into consideration it's an e2e test we cannot control every single aspect of the process, what we need to ensure is the final state, the agent is enrolled and monitoring is working, what is being done by this test.